torchgeo/datasets/everwatch.py

```diff
@@ -133,7 +134,8 @@ def __len__(self) -> int:
         Returns:
             length of the dataset
         """
-        return len(self.annot_df.index.levels[0])
+        index = cast(pd.MultiIndex, self.annot_df.index)
```

Comment:
Mypy cannot understand that our DataFrame went from Index to MultiIndex. I tried to find ways to avoid this dynamic typing, but I'm not even sure how the pandas type hints differentiate between these two types of DataFrames anyway.